projects/scripts/fmc_constr_generator.tcl

```diff
@@ -0,0 +1,276 @@
+        for {set j 1} {$j < [expr [llength $line_c] / 5]} {incr j} { 
+          if {[string compare [lindex $line_e [expr $i*6]] [lindex $line_c [expr $j*5]]] == 0} {
+            if {[string compare [lindex $line_c [expr $j*5+2]] "#N/A"] != 0} {
+			  if {[string match $carrier vcu118] && [string match "*[lindex $line_c [expr $j*5]]*" "B20B21"]} { 
+			    if {[string compare [string replace [lindex $line_c [expr $j*5+3]] 0 [string last "_" [lindex $line_c [expr $j*5+3]]] ""] $rx_bank] != 0} {continue}
+			  }
+			  if {[string match $carrier vcu118] && [string match "*[lindex $line_c [expr $j*5]]*" "D4D5"]} { 
+			    if {[string compare [string replace [lindex $line_c [expr $j*5+3]] 0 [string last "_" [lindex $line_c [expr $j*5+3]]] ""] $tx_bank] != 0} {continue}
+			  }
+              set spaces_0 ""
+              set spaces_1 ""
+              set spaces_2 ""
+              set spaces_3 ""
+              for {set k 0} {$k < [expr $col0_max - [string length [lindex $line_c [expr $j*5]]]]} {incr k} {append spaces_0 " "}
+              for {set k 0} {$k < [expr $col1_max - [string length [lindex $line_c [expr $j*5+1]]]]} {incr k} {append spaces_1 " "}
+              for {set k 0} {$k < [expr $col2_max - [string length [lindex $line_c [expr $j*5+2]]]]} {incr k} {append spaces_2 " "}
+              for {set k 0} {$k < [expr $col3_max - [string length [lindex $line_e [expr $i*6+3]]]]} {incr k} {append spaces_3 " "} 
+              if {[string compare [lindex $line_e [expr $i*6+4]] "#N/A"] != 0} {
+                set io_standard "$spaces_2 IOSTANDARD [lindex $line_e [expr $i*6+4]]"
+                if {[string compare [lindex $line_e [expr $i*6+5]] "#N/A"] != 0} {
+                  append io_standard " [lindex $line_e [expr $i*6+5]]"
+                }
+              } else {
+                  set io_standard ""
+              }
+              set io_standard [string map -nocase {"," " "} $io_standard]
+              if {[string length $io_standard] > 0} {
+			    for {set k 0} {$k < [expr $io_standard_max + [string length $spaces_2] + 12 - [string length $io_standard]]} {incr k} {append spaces_3 " "}
+                puts $constr_file "set_property -dict \{PACKAGE_PIN [lindex $line_c [expr $j*5+2]]$io_standard\} \[get_ports [lindex $line_e [expr $i*6+3]]\]$spaces_3 ; ## [lindex $line_c [expr $j*5]]$spaces_0  [lindex $line_c [expr $j*5+1]] $spaces_1 [lindex $line_c [expr $j*5+3]]" 
+			  } else {
+			      for {set k 0} {$k < [expr $io_standard_max + 12 - [string length $io_standard]]} {incr k} {append spaces_3 " "}
+			      puts $constr_file "set_property -dict \{PACKAGE_PIN [lindex $line_c [expr $j*5+2]]\}$spaces_2 \[get_ports [lindex $line_e [expr $i*6+3]]\]$spaces_3 ; ## [lindex $line_c [expr $j*5]]$spaces_0  [lindex $line_c [expr $j*5+1]] $spaces_1 [lindex $line_c [expr $j*5+3]]" 
+		        } 
+            }
+          }
+        }
+      }
+    }
+  close $constr_file
+  add_files -fileset constrs_1 -norecurse fmc_constr.xdc
```

Comment:
This is also a command that considers we have an opened vivado project. Having it as an extra feature is the way to go, in my opinion.